### PR TITLE
Exploration

### PR DESCRIPTION
**Pull request by tdimnet**
Create the tables of runs by:
- Creating a new service for the AJAX Request.
- Adding a new npm library for handling the data displayed.
- Adding custom styles to the table.

### CODE REVIEW

File: webserver/client/src/components/RunsList/index.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+                resizable={false}
+                className="-striped"
+            />
+
+        );
+    }
+};
+
+
+export default RunsTasks;
```
Comment (leopoldhoudin) [inline webserver/client/src/components/RunsList/index.js]: Shouldn't be?
```js
export default RunsList;
```

File: webserver/client/src/components/RunsList/index.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import React from 'react';
```
Comment (leopoldhoudin) [inline webserver/client/src/components/RunsList/index.js]: `prop-types` ?

File: webserver/client/src/components/RunsList/index.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import ReactDOM from 'react-dom';
+import ReactTable from 'react-table';
+import { Link } from 'react-router-dom';
+import FontAwesome from 'react-fontawesome';
+
+import StatusTag from '../StatusTag';
+
+import '../../../node_modules/react-table/react-table.css';
```
Comment (leopoldhoudin) [inline webserver/client/src/components/RunsList/index.js]: `import` to move in `style.scss`

File: webserver/client/src/components/RunsList/index.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+import './style.scss';
+
+
+class RunsTasks extends React.Component {
```
Comment (leopoldhoudin) [inline webserver/client/src/components/RunsList/index.js]: add explicit default `constructor`

File: webserver/client/src/components/RunsList/index.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    render() {
+
+        const columns = [
```
Comment (leopoldhoudin) [inline webserver/client/src/components/RunsList/index.js]: couldn't you move that into the constructor as `this.columns` ?
that would slightly optimize rendering (i.e. you are now building again the whole columns definition each time you render).

File: webserver/client/src/components/RunsList/index.js [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+            {
+                header: 'Status',
+                accessor: 'return_code',
+                render: ({value}) => (<StatusTag return_code={value} />)
+            },
+            {
+                header: () => <span className="icon is-small">Task<FontAwesome name="sort" /><FontAwesome name="sort-asc" /><FontAwesome name="sort-desc" /></span>,
```
Comment (leopoldhoudin) [inline webserver/client/src/components/RunsList/index.js]: all you headers are the same: refactorize

File: webserver/client/src/components/RunsList/index.js [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+                id: 'taskName',
+                accessor: (run) => run.task.name
+            },
+            {
+                header: () => <span className="icon is-small">Start<FontAwesome name="sort" /><FontAwesome name="sort-asc" /><FontAwesome name="sort-desc" /></span>,
+                id: 'startedAt',
+                accessor: (run) => run.started_at.substring(0, 19),
```
Comment (leopoldhoudin) [inline webserver/client/src/components/RunsList/index.js]: Use `DateTimeService` for formatting? (i.e. `DateTimeService. toString(datetime)`)

File: webserver/client/src/components/RunsList/style.scss [hunk 8]
```diff
@@ -0,0 +1,90 @@
+.table {
+    width: 90%;
+}
+
+// Remove the default stylesheets fron the ReactTable Component
+.ReactTable {
+    border: none;
+    color: #7a7a7a;
```
Comment (leopoldhoudin) [inline webserver/client/src/components/RunsList/style.scss]: use variables for colors

File: webserver/client/src/components/RunsTasks/index.js [hunk 9]
```diff
@@ -0,0 +1,55 @@
+import React from 'react';
```
Comment (tdimnet) [inline webserver/client/src/components/RunsTasks/index.js]: This class has been removed.
commit code: a72c5cf48db9f792b49100486d97e02d9913f71a

File: webserver/client/src/pages/Runs/index.js [hunk 10]
```diff
@@ -1,29 +1,64 @@
 import React from 'react';
+import ReactDOM from 'react-dom';
+
+import ReactTable from 'react-table';
+
+import StatusTag from 'components/StatusTag';
```
Comment (tdimnet) [inline webserver/client/src/pages/Runs/index.js]: Useless statements removed.
Commit code: 484d7f443e71334cefb34cf795ba0ff8bd556ba0

File: webserver/client/src/pages/Runs/index.js [hunk 11]
```diff
@@ -1,29 +1,64 @@
...
+
+import RunsList from 'components/RunsList';
+import RunsService from 'services/RunsService';
+
+import PropTypes from 'prop-types';
+
+import '../../../node_modules/react-table/react-table.css';
```
Comment (tdimnet) [inline webserver/client/src/pages/Runs/index.js]: The first import (PropTypes) is needed => we use this at the bottom of the index.js page.
The second one has been removed.
Commit code: ad50324fe0add2414cc909d87a909a58690a1859

File: webserver/client/src/pages/Runs/style.scss [hunk 12]
```diff
@@ -0,0 +1,4 @@
+.ReactTable {
```
Comment (tdimnet) [inline webserver/client/src/pages/Runs/style.scss]: Also remove some unnecessary style inside the style.scss of RunsList.
Commit code: c115354116bd87d4daf9b219811dc691db967f3b

File: webserver/client/src/components/RunsList/index.js (hunk 2 above)
Comment (tdimnet) [inline webserver/client/src/components/RunsList/index.js]: The prop-types is already defined inside the Runs page: the API is called from this file and no Runs/List/index.js

File: webserver/client/src/components/RunsList/index.js (hunk 3 above)
Comment (tdimnet) [inline webserver/client/src/components/RunsList/index.js]: File moved from the RunsList/index.js
Commit code: 96cae7c7e65d86bb817d020705a218dec109386c

File: webserver/client/src/components/RunsList/index.js (hunk 6 above)
Comment (tdimnet) [inline webserver/client/src/components/RunsList/index.js]: Created a custom component called HeaderTag which manages the headers.
Commit code: f39366ca16a81a74652a2ab752a8c2e65132a2b0

File: webserver/client/src/pages/Runs/index.js [hunk 13]
```diff
@@ -1,29 +1,59 @@
...
+                    </div>
 
-                    </section>
+                    <RunsList runs={this.state.runs} />
 
                 </div>);
     }
 }
 
+Runs.propTypes = {
```
Comment (leopoldhoudin) [inline webserver/client/src/pages/Runs/index.js]: - [ ] `Runs` class does not have `runs` `props`, to remove

File: webserver/client/src/components/RunsList/index.js (hunk 2 above)
Comment (leopoldhoudin) [inline webserver/client/src/components/RunsList/index.js]: No, the `props` for `RunsList` should be defined in the component's file, i.e. here.
`Runs` component does not have a `runs` prop.

File: webserver/client/src/components/HeaderTag/index.js [hunk 14]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
```
Comment (leopoldhoudin) [inline webserver/client/src/components/HeaderTag/index.js]: Nice 👍 
Though is not a Tag...
Rename to `SortableHeaderCell`

Comment (tdimnet) [inline webserver/client/src/components/HeaderTag/index.js]: New name applied.
Commit number: 4f3d676ad55582fced39ae0e6f77c0fcdd76b860

File: webserver/client/src/components/RunsList/index.js (hunk 4 above)
Comment (leopoldhoudin) [inline webserver/client/src/components/RunsList/index.js]: rename class into `RunsList`

**Comment by tdimnet** [comment]
Change the sass file by creating new variables (The previous changes comment has been removed)

File: webserver/client/src/components/RunsList/index.js (hunk 5 above)
Comment (tdimnet) [inline webserver/client/src/components/RunsList/index.js]: Done.
Commit code: 536875055ddb45d5b435009875acefadafcc5072

File: webserver/npm-debug.log [hunk 15]
```diff
@@ -0,0 +1,20 @@
+0 info it worked if it ends with ok
```
Comment (tdimnet) [inline webserver/npm-debug.log]: Done.
Commit code: ab2a0f9319cab229dcc1877b62c428181da8229b

File: webserver/client/src/components/RunsList/index.js (hunk 1 above)
Comment (tdimnet) [inline webserver/client/src/components/RunsList/index.js]: Done.
Commit Code: bf7287c47f542bb09aee447e845ea92112f7951c

File: webserver/client/src/pages/Runs/index.js (hunk 13 above)
Comment (tdimnet) [inline webserver/client/src/pages/Runs/index.js]: Done.
Commit Code: f0e86c3a9cbe8f579ad3709166b909f826b3b2ea

File: webserver/client/src/components/RunsList/index.js (hunk 1 above)
Comment (leopoldhoudin) [inline webserver/client/src/components/RunsList/index.js]: You still need to put the prop-types overhere

File: webserver/client/src/components/SortableHeaderCell/index.js [hunk 16]
```diff
@@ -0,0 +1,21 @@
...
+            <span className="icon is-small">
+                {headerName}
+                <FontAwesome name="sort" />
+                <FontAwesome name="sort-asc" />
+                <FontAwesome name="sort-desc" />
+            </span>)
+    }
+};
+
+export default SortableHeaderCell;
```
Comment (leopoldhoudin) [inline webserver/client/src/components/SortableHeaderCell/index.js]: you need to define your prop-types overhere as well

Comment (tdimnet) [inline webserver/client/src/components/SortableHeaderCell/index.js]: Done : create two prop-types (only for the data displayed within the table, not for the full elements of the object).
Commit Code: 1aeffcae636a12de060a7785496ee21d09f0bf83

File: webserver/client/src/components/RunsList/index.js [hunk 17]
```diff
@@ -49,12 +51,15 @@ class RunsList extends React.Component {
...
             },
-        ]
+        ];
     }
 
 
     render() {
 
+
+        console.log(this.props.runs);
```
Comment (leopoldhoudin) [inline webserver/client/src/components/RunsList/index.js]: `console.log`

File: webserver/client/package.json [hunk 18]
```diff
@@ -2,7 +2,9 @@
   "name": "cronos-client",
   "version": "0.0.0",
   "main": "main.js",
-  "dependencies": {},
+  "dependencies": {
```
Comment (tdimnet) [inline webserver/client/package.json]: Done.
Code commit: 47624297f1abee07ce7ef14e1049aa14615a805a

File: webserver/client/src/components/RunsList/index.js (hunk 17 above)
Comment (tdimnet) [inline webserver/client/src/components/RunsList/index.js]: Done.
Code commit: 47624297f1abee07ce7ef14e1049aa14615a805a

**Comment by leopoldhoudin** [comment]
@tdimnet c'est good :-)

**Pull request merged**